lib/medtronic/processData.js

```diff
@@ -236,12 +233,21 @@ var buildWizardRecords = function (records) {
       bgTarget.high = struct.extractByte(record.body,12);
       recommended.carb = struct.extractBEShort(record.body,6) / cfg.settings.strokesPerUnit;
 
-      var correctionRaw = [twosComplement(struct.extractByte(record.body,5) & BITMASKS.RIGHT_FOUR)];
-      correctionRaw = correctionRaw.concat(twosComplement(struct.extractByte(record.body,7)));
+      var correctionRaw = [twosComplement(struct.extractByte(record.body,5) & BITMASKS.RIGHT_FOUR, 8)];
+      correctionRaw = correctionRaw.concat(twosComplement(struct.extractByte(record.body,7), 8));
       recommended.correction = struct.extractBEShort(correctionRaw,0) / cfg.settings.strokesPerUnit;
 
       recommended.net = struct.extractBEShort(record.body,11) / cfg.settings.strokesPerUnit;
       iob =  struct.extractBEShort(record.body,9) / cfg.settings.strokesPerUnit;
+
+      units = 'mg/dL'; // TODO: figure out these units when we get hold of pump model with smaller record format
+    }
+
+    if(units == 'mmol/L') {
```

Comment:
Yeah, it will be sent as 78 if the value is 7.8 mmol/L. Not sure how exactly floating point math will cause issues in this specific instance, but I've seen enough strange things with floating point to agree that it's better to play it safe.